### DISCUSSION TITLE
Improved difference image to better show subtle differences.

### DISCUSSION THREAD

**Discussion by CraigSiemens** [pointfreeco/swift-snapshot-testing#515] [Ideas]
Would it be possible to normalize the difference image that's created when a snapshot fails? That way if a snapshot only has really subtle differences, they can still be seen in the difference image.

---

We're updating our app to build with Xcode12 and iOS 15. A bunch if our image snapshot tests have started failing, presumably due to rendering differences between iOS 14 and 15. When we look at the difference image that generated, they look blank.

This is an extreme example, but does show it off.
<img width='25%' src='https://user-images.githubusercontent.com/680442/133168709-851dcf12-5cfb-47f7-afff-c08abbfce8bc.png'>

If I take the image, open in preview > adjust colors > auto levels, it brightens up the areas of the image that are different.

This is cropped to show the important part.
![difference_3_DDFA5885-E945-494C-A9CA-AE69DD64940E copy](https://user-images.githubusercontent.com/680442/133169059-8c17f667-d0e3-44f2-8f7e-9687b0e73443.png)

It'd be nice if the output image could be normalized to brighten up those ares without needing to use another tool.

---

Another option could be to do something like the `compare` command from ImageMagick which shows a faded out version of the reference image, then colors in red any part that is different.

This is from running
`compare reference.png failure.png output.png`

![output.png](https://user-images.githubusercontent.com/680442/133169062-56c8adde-acc4-4823-9649-10cf37a49296.png)

**Comment by mbrandonw** [comment]
Hey @CraigSiemens, this is a great idea, but I'm going to move it over to our discussions because it isn't really a bug with the library.

**Comment by penelopearaujo** [comment]
I'm facing the same issue, lots of failed tests and blank diffs after the app was updated with iOS 15 SDK. Any thoughts on this? 
(I'm already checking with `preview > adjust colors > auto levels`, which has been very helpful! Thanks!)
